coalib/core/CircularDependencyError.py

```diff
@@ -12,8 +13,24 @@ def __init__(self, node=None):
             The node that was encountered twice and closes the dependency
             circle.
         """
-        message = 'Circular dependency detected.'
+        prefix = 'Circular dependency detected'
         if node is not None:
-            message += ' {0!r} -> ... -> {0!r}'.format(node)
+            message = '{prefix}: {node!r} -> ... -> {node!r}'.format(
+                prefix=prefix, node=node)
+        else:
+            message = prefix + '.'
+        return cls(message)
 
-        Exception.__init__(self, message)
+    @classmethod
+    def for_bears(cls, bears):
+        """
+        Creates the CircularDependencyError with a helpful message about the
+        dependency.
+
+        :param bears:
+            sequence of bears, for which a circular dependency was detected.
+        """
+        bear_names = [bear.name for bear in bears]
+
+        return cls('Circular dependency detected: {names}'.format(
+            names=' -> '.join(bear_names)))
```

Comment:
> Then what is the need for having special implementations only used at a single place? More flexible functions + less code to maintain ￼

That's a good point. Since it's only used at a single place, I've changed it as suggested by you.